src/bag3_testbenches/measurement/char/__init__.py

```diff
@@ -216,40 +223,70 @@ def compute_passives(meas_results: Mapping[str, Any], passive_type: str) -> Mapp
     )
     if passive_type == 'cap':
         results['cc'] = estimate_cap(freq, yc)
-        results['r_series'] = np.mean(1 / yc).real
+        results['r_series'] = (1 / yc).mean(axis=-1).real
     elif passive_type == 'res':
         results['c_parallel'], results['res'] = estimate_esd(freq, yc)
-
-        warr = 2 * np.pi * freq
-        z_meas = 1 / (yc + (ypp * ypm) / (ypp + ypm))
-        if results['cpp'] == 0 or results['cpm'] == 0:
+        if np.mean(results['cpp']) == 0 or np.mean(results['cpm']) == 0:
             cp_est = 0
         else:
             cp_est = 1 / (1 / results['cpp'] + 1 / results['cpm'])
-        z_est = 1 / (1 / results['res'] + 1j * warr * (results['c_parallel'] + cp_est))
-        plt.semilogx(freq, np.abs(z_meas), label='Measured')
-        plt.semilogx(freq, np.abs(z_est), label='Estimated')
-        plt.xlabel('Frequency (in Hz)')
-        plt.ylabel('Value')
-        plt.legend()
-        plt.grid()
-        plt.show()
+        if results['c_parallel'].all() > 0:
```

Comment:
Agreed on the if condition. Do we need an else? Raise an error?

Reply:
The `else` condition is to not compute the BW which is the default, which happens in schematic simulations where there won't be a `c_parallel`.